File: Services/PoolValue.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace AqualogicJumper.Services
{
    public class PoolValue
    {
        public int Id => (int) Name;
        [JsonConverter(typeof(StringEnumConverter))]
        public PoolValueName Name { get; set; }
        public string Text { get; set; }
        public object Value { get; set; }
        [JsonConverter(typeof(StringEnumConverter))]
        public ValueUnit? Unit { get; set; }
        public bool Modified { get; set; }
    }
}

File: Content/portPool/poolValue.ts
export interface PoolValue {
    id: number;
    name: PoolValueName;
    text: string;
    value: any;
    unit: ValueUnit | null;
    modified: boolean;
}
export enum PoolValueName {
    Unknown,
    PoolTemp,
    SpaTemp,
    AirTemp,
    PoolChlorinator,
    SpaChlorinator,
    SaltLevel,
    CheckSystem,
    HeatPump,
    HeaterAuto
}

export enum ValueUnit {
    DegreesF,
    DegreesC,
    Percentage,
    GramsLiter,
    PartsPerMillion,
    String,
    AutoMode
}